### ISSUE TITLE
Connection problem with Android 4.3 or later.

### ISSUE THREAD

**Issue by zuma-silent** [xelerance/xl2tpd#44]
I run openswan 2.6.38 and xl2tpd-1.3.6 on the server and I try to connect with Android 4.4.4 using IPSec/L2TP psk.
The VPN tunnel is established but then it seems that xl2tpd terminates the pppd.

LOGS
_xl2tpd[7232]: handle_avps: handling avp\'s for tunnel 37735, call 0_
_xl2tpd[7232]: message_type_avp: message type 4 (Stop-Control-Connection-Notification)_
_xl2tpd[7232]: assigned_tunnel_avp: using peer\'s tunnel 23332_
_xl2tpd[7232]: result_code_avp: avp is incorrect size.  8 < 10_
_xl2tpd[7232]: handle_avps: Bad exit status handling attribute 1 (Result Code) on mandatory packet._
_xl2tpd[7232]: handle_packet: bad AVP handling!_
_xl2tpd[7232]: network_thread: bad packet_
_xl2tpd[7232]: build_fdset: closing down tunnel 37735_
_xl2tpd[7232]: Terminating pppd: sending TERM signal to pid 7366_

The configuration in ipsec.conf is:
conn l2tp-psk
                  _authby=secret_
                  _pfs=no_
                  _auto=add_
                  _rekey=no_
                  _type=transport_
                  _left=%defaultroute_
                  _leftprotoport=17/1701_
                  _right=%any_
                  _rightprotoport=17/%any_
                  _rightsubnet=vhost:%priv,%no_
                  _forceencaps=yes_

And xl2tpd.conf is:

  [global]
  listen-addr = x.x.x.x

  [lns default]
  ip range = 10.254.253.2-10.254.253.250
  local ip = 10.254.253.1
  assign ip = yes
  require chap = yes
  refuse pap = yes
 require authentication = yes
 name = OpenswanVPN
 ppp debug = yes
 pppoptfile = /etc/ppp/options.xl2tpd
 length bit = yes

L2tp is working with iOS and Android of previous version.
The problem appeared from 4.3.x version.
Do you have any idea?
Can it be an Android issue?

Thank you

**Comment by Show-vars** [comment]
yes, some problem on 4.4.2
```
xl2tpd[1312]: check_control: Received out of order control packet on tunnel -1 (got 1, expected 0)
xl2tpd[1312]: handle_packet: bad control packet!
xl2tpd[1312]: Maximum retries exceeded for tunnel 54606.  Closing.
xl2tpd[1312]: Connection 46225 closed to *.*.*.*, port 61626 (Timeout)
xl2tpd[1312]: Unable to deliver closing message for tunnel 54606. Destroying anyway.
```

**Comment by zuma-silent** [comment]
The error has been resolved in nexus android 5.1.
